src/Collectors/ScheduledTaskCollector.php

```diff
@@ -26,27 +26,21 @@ public function registerEventListeners(): void
     {
         $this->app->events->listen(ScheduledTaskStarting::class, function (ScheduledTaskStarting $event) {
             $transaction_name = $this->getTransactionName($event);
-            if (!$transaction_name) {
-                return;
-            }
-
-            $transaction = $this->getTransaction($transaction_name);
-            if ($transaction) {
-                // Somehow, a transaction with the same name has already been created.
-                // If so, ignore this job, otherwise the agent will throw an exception.
-                return;
+            if ($transaction_name) {
+                $transaction = $this->getTransaction($transaction_name);
+                if (!$transaction) {
+                    $transaction = $this->startTransaction($transaction_name);
+                    $this->addMetadata($transaction);
+                }
             }
-
-            $transaction = $this->startTransaction($transaction_name);
-            $this->addMetadata($transaction);
         });
 
         $this->app->events->listen(ScheduledTaskSkipped::class, function (ScheduledTaskSkipped $event) {
             $transaction_name = $this->getTransactionName($event);
             if ($transaction_name) {
                 $transaction = $this->getTransaction($transaction_name);
                 if ($transaction) {
-                    $this->stopTransaction($transaction_name, 200);
+                    $this->stopTransaction($transaction_name, $event->task->exitCode);
```

Comment:
same as commands? I mean it makes sense, just I haven't paid attention before.

Reply:
Yeap, at the end the task wraps a command, so the exit code is available as well.